x-pack/plugins/transform/server/routes/api/transforms_nodes.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPopulatedObject } from '../../../common/utils/object_utils';
+
+import { RouteDependencies } from '../../types';
+
+import { addBasePath } from '../index';
+
+import { wrapError, wrapEsError } from './error_utils';
+
+interface NodesAttributes {
+  attributes: {
+    'transform.node': string;
+  };
+}
+type Nodes = Record<string, NodesAttributes>;
+
+const isNodes = (arg: unknown): arg is Nodes => {
+  return (
+    isPopulatedObject(arg) &&
+    Object.values(arg).every(
+      (node) =>
+        isPopulatedObject(node) &&
+        {}.hasOwnProperty.call(node, 'attributes') &&
+        isPopulatedObject(node.attributes) &&
+        {}.hasOwnProperty.call(node.attributes, 'transform.node')
```

Comment:
i think this check is too strict, `transform.node` may be missing entirely from one of the nodes, in which case the whole check will fail and count will be returned as 0

Reply:
Good catch and thinking about multiple nodes! I fixed the type guard and added a unit test in cf49a77.